mutiny-core/src/gossip.rs

```diff
@@ -64,40 +83,82 @@ async fn get_gossip_data(
     log_debug!(logger, "Got network graph, getting scorer...");
 
     // Get the probabilistic scorer
-    let scorer = match storage.get_data::<String>(PROB_SCORER_KEY)? {
-        Some(prob_scorer_str) => {
-            let prob_scorer_bytes: Vec<u8> = Vec::from_hex(&prob_scorer_str)?;
-            let mut readable_bytes = lightning::io::Cursor::new(prob_scorer_bytes);
-            let params = ProbabilisticScoringDecayParameters::default();
-            let args = (params, Arc::clone(&network_graph), Arc::clone(&logger));
-            ProbScorer::read(&mut readable_bytes, args)
-        }
-        None => {
-            let gossip = Gossip {
-                last_sync_timestamp,
-                network_graph,
-                scorer: None,
-            };
-            return Ok(Some(gossip));
+    let scorer = match (remote_scorer_url, &auth_client) {
+        (Some(url), Some(client)) => match get_remote_scorer_bytes(client, &url).await {
+            Ok(scorer_bytes) => {
+                let mut readable_bytes = lightning::io::Cursor::new(scorer_bytes);
+                let params = ProbabilisticScoringDecayParameters::default();
+                let args = (params, Arc::clone(&network_graph), Arc::clone(&logger));
+                match ProbScorer::read(&mut readable_bytes, args) {
+                    Ok(scorer) => {
+                        log_debug!(logger, "retrieved remote scorer");
+                        Some(scorer)
+                    }
+                    Err(_) => {
+                        log_error!(
+                            logger,
+                            "failed to parse remote scorer, failing back to local"
+                        );
+                        get_scorer(storage, network_graph.clone(), logger.clone()).await?
+                    }
+                }
+            }
+            Err(_) => {
+                log_error!(
+                    logger,
+                    "failed to retrieve remote scorer, failing back to local"
+                );
+                get_scorer(storage, network_graph.clone(), logger.clone()).await?
+            }
+        },
+        _ => {
+            log_debug!(logger, "only getting local scorer");
+            get_scorer(storage, network_graph.clone(), logger.clone()).await?
         }
     };
 
-    if let Err(e) = scorer.as_ref() {
+    if scorer.is_none() {
         log_warn!(
             logger,
-            "Could not read probabilistic scorer from database: {e}"
+            "Could not read probabilistic scorer from database or remote server"
         );
     }
 
     let gossip = Gossip {
         last_sync_timestamp,
         network_graph,
-        scorer: scorer.ok(),
+        scorer,
     };
 
     Ok(Some(gossip))
 }
 
+/// Scorer is the scorer that gets pulled remotely
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Scorer {
+    pub value: Vec<u8>,
```

Comment:
Would be better if we encode as hex or even better base64 to reduce bandwidth.

Reply:
Encoding will increase bandwidth

Reply:
It's written as [1, 2, 3,..] in the json so hex is more effienct

Reply:
Damn you're right actually. I guess I can get rid of all the serialization completely then and just deal with raw bytes, no object?

Reply:
Yeah that'd be best I guess

Reply:
I spent far too long messing with stupid byte responses that wouldn't properly decode, kept the json but swapped for base64.